### PR TITLE
removing assert statement used outside of tests

### LINKED ISSUE #1903: Assert statement used outside of tests
Since assert provides an easy way to check some condition and fail execution, it’s very common for developers to use it to check validity. But when the Python interpreter is invoked with the -O (optimize) flag, the assert statements are removed from the bytecode.

Files:
```
mopidy/audio/utils.py
mopidy/core/actor.py
mopidy/core/playback.py
mopidy/core/tracklist.py
mopidy/ext.py
```
are using assert statements.

### PR DESCRIPTION
**Pull request by asmijafar20**
fix #1903

### CODE REVIEW

File: mopidy/core/playback.py [hunk 1]
```diff
@@ -301,7 +301,8 @@ def play(self, tl_track=None, tlid=None):
 
         if tl_track is not None:
             # TODO: allow from outside tracklist, would make sense given refs?
-            assert tl_track in self.core.tracklist.get_tl_tracks()
+            if (tl_track not in self.core.tracklist.get_tl_tracks()):
```
Comment (jodal) [inline mopidy/core/playback.py:304]: Parenthesis are not needed here:

```suggestion
            if tl_track not in self.core.tracklist.get_tl_tracks():
```

File: mopidy/core/tracklist.py [hunk 2]
```diff
@@ -480,15 +480,18 @@ def move(self, start, end, to_position):
...
-        assert start < end, "start must be smaller than end"
-        assert start >= 0, "start must be at least zero"
-        assert end <= len(
-            tl_tracks
-        ), "end can not be larger than tracklist length"
-        assert to_position >= 0, "to_position must be at least zero"
-        assert to_position <= len(
-            tl_tracks
-        ), "to_position can not be larger than tracklist length"
+        if not start < end:
```
Comment (jodal) [inline mopidy/core/tracklist.py:483]:
```suggestion
        if start >= end:
```

File: mopidy/core/tracklist.py [hunk 3]
```diff
@@ -480,15 +480,18 @@ def move(self, start, end, to_position):
...
+            raise AssertionError("start must be smaller than end")
+        if not start >= 0:
```
Comment (jodal) [inline mopidy/core/tracklist.py:485]:
```suggestion
        if start < 0:
```

File: mopidy/core/tracklist.py [hunk 4]
```diff
@@ -480,15 +480,18 @@ def move(self, start, end, to_position):
...
+            raise AssertionError("start must be at least zero")
+        if not end <= len(tl_tracks):
```
Comment (jodal) [inline mopidy/core/tracklist.py:487]:
```suggestion
        if end > len(tl_tracks):
```

File: mopidy/core/tracklist.py [hunk 5]
```diff
@@ -480,15 +480,18 @@ def move(self, start, end, to_position):
...
+            raise AssertionError("end can not be larger than tracklist length")
+        if not to_position >= 0:
```
Comment (jodal) [inline mopidy/core/tracklist.py:489]:
```suggestion
        if to_position < 0:
```

File: mopidy/core/tracklist.py [hunk 6]
```diff
@@ -480,15 +480,18 @@ def move(self, start, end, to_position):
...
+            raise AssertionError("to_position must be at least zero")
+        if not to_position <= len(tl_tracks):
```
Comment (jodal) [inline mopidy/core/tracklist.py:491]:
```suggestion
        if to_position > len(tl_tracks):
```

File: mopidy/core/tracklist.py [hunk 7]
```diff
@@ -532,15 +535,18 @@ def shuffle(self, start=None, end=None):
 
         # TOOD: use validation helpers?
         if start is not None and end is not None:
-            assert start < end, "start must be smaller than end"
+            if not start < end:
```
Comment (jodal) [inline mopidy/core/tracklist.py:538]:
```suggestion
            if start >= end:
```

File: mopidy/core/tracklist.py [hunk 8]
```diff
@@ -532,15 +535,18 @@ def shuffle(self, start=None, end=None):
...
+                raise AssertionError("start must be smaller than end")
 
         if start is not None:
-            assert start >= 0, "start must be at least zero"
+            if not start >= 0:
```
Comment (jodal) [inline mopidy/core/tracklist.py:542]:
```suggestion
            if start < 0:
```

File: mopidy/core/tracklist.py [hunk 9]
```diff
@@ -532,15 +535,18 @@ def shuffle(self, start=None, end=None):
...
+                raise AssertionError("start must be at least zero")
 
         if end is not None:
-            assert end <= len(tl_tracks), (
-                "end can not be larger than " + "tracklist length"
-            )
+            if not end <= len(tl_tracks):
```
Comment (jodal) [inline mopidy/core/tracklist.py:546]:
```suggestion
            if end > len(tl_tracks):
```

File: mopidy/core/playback.py (hunk 1 above)
Comment (jodal) [reply mopidy/core/playback.py:304]: This was also caught by the Black formatting check at CircleCI: https://circleci.com/gh/mopidy/mopidy/1953

**Comment by jodal** [comment]
Thanks :-)

**Comment by asmijafar20** [comment]
Your welcome :)

**Pull request merged**